04-onion-routing.md

```diff
@@ -269,7 +272,81 @@ The writer:
 The reader:
   - MUST return an error if `amt_to_forward` or `outgoing_cltv_value` are not present.
 
-The requirements for the contents of these fields are specified [above](#legacy-hop_data-payload-format).
+The requirements for the contents of these fields are specified [above](#legacy-hop_data-payload-format)
+and [below](#basic-multi-part-payments).
+
+### Basic Multi-Part Payments
```

Comment:
I'm not sure this should be in Bolt 4. Apart from the TLV that signals support, all the AMP stuff isn't really related to onion encryption or routing. Maybe AMP should have its own Bolt? Not sure but if it becomes bigger (when we add more advanced AMP features) it will feel messy to have it here, won't it?

Reply:
Yeah, BOLT4 is where all the "node" semantics ended up, this just continues the trend.

Separating the handling into a new bolt is probably a good idea, but a fairly large and invasive job which I'd prefer to happen while we're not actively churning the spec!